### PR TITLE
Add `TabBar.tabAlignment` migration guide

### PR DESCRIPTION
**Pull request by TahaTesser**
fixes https://github.com/flutter/flutter/issues/126920

[TabBar.tabAlignment](https://master-api.flutter.dev/flutter/material/TabBar/tabAlignment.html) was introduced in https://github.com/flutter/flutter/pull/125036

This PR adds a  migration guide for `TabBar.tabAlignment`.

Related https://github.com/flutter/flutter/pull/125974

## Presubmit checklist

- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/main/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

File: src/release/breaking-changes/tab-alignment.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+description: Introducing the TabBar.tabAlignment property.
+---
+
+## Summary
+
+Using `TabBar.tabAlignment` to customize the alignment of tabs in a `TabBar`.
+
+## Context
+
+The `TabBar.tabAlignment` property is introduced to customize the alignment of tabs in a Material 3 `TabBar`. The `TabAlignment` enum has the following values:
```
Comment (Piinks) [inline src/release/breaking-changes/tab-alignment.md:12]:
```suggestion
The `TabBar.tabAlignment` property was introduced to customize the alignment of tabs in a Material 3 `TabBar`. The `TabAlignment` enum has the following values:
```

File: src/release/breaking-changes/tab-alignment.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+* `TabAlignment.start`: Aligns the tabs to the start of the scrollable `TabBar`.
+* `TabAlignment.startOffset`: Aligns the tabs to the start of the scrollable `TabBar` with an offset of `52.0` pixels.
+* `TabAlignment.center`: Aligns the tabs to the center of the `TabBar`.
+* `TabAlignment.fill`: Aligns the tabs to the start and stretches the tabs to fill the fixed `TabBar`.
```
Comment (Piinks) [inline src/release/breaking-changes/tab-alignment.md:17]: Should these descriptions also mention which ones work with `isScrollable` and which ones don't?

**Review by Piinks — COMMENTED** [review]
This is great! Just a couple of small nits

File: src/release/breaking-changes/tab-alignment.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+When `ThemeData.useMaterial3` are set to `true`, the default tab alignment for a scrollable `TabBar` is `TabAlignment.startOffset`. This can be changed by setting the `TabBar.tabAlignment` property.
+
+## Description of change
+
+When the `TabBar.isScrollable` and `ThemeData.useMaterial3` are set to `true`, the tabs in a scrollable `TabBar` defaults to `TabAlignment.startOffset` which aligns the tabs to the start of the scrollable `TabBar` with an offset of `52.0` pixels. This is a change from the previous behavior, where the tabs were aligned to the start of the scrollable `TabBar` when there are more tabs to display than the available width.
+
+## Migration guide
+
+A Material 3 scrollable `TabBar` uses `TabAlignment.startOffset` as the default tab alignment which aligns the tabs to the start of the scrollable `TabBar` with an offset of `52.0` pixels.
```
Comment (atsansone) [inline src/release/breaking-changes/tab-alignment.md:27]: **issues:** 
- Write as shorter sentences per [GSG](https://developers.google.com/style/accessibility#ease-of-reading).
- Use [semantic breaks](https://sembr.org).

```suggestion
A Material 3 scrollable `TabBar` uses `TabAlignment.startOffset` as the default tab alignment.
This aligns the tabs to the start of the scrollable `TabBar` with an offset of `52.0` pixels.
```

File: src/release/breaking-changes/tab-alignment.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+To align the tabs to the start of the scrollable `TabBar`, without an offset of `52.0` pixels, set `TabBar.tabAlignment` to `TabAlignment.start`.
+The following code snippets show how to use `TabBar.tabAlignment` to align the tabs to the start of the scrollable `TabBar`:
```
Comment (atsansone) [inline src/release/breaking-changes/tab-alignment.md:30]: **issues:** 
- Write as shorter sentences per [GSG](https://developers.google.com/style/accessibility#ease-of-reading).
- Use [semantic breaks](https://sembr.org).

```suggestion
To align the tabs to the start of the scrollable `TabBar`, set `TabBar.tabAlignment` to `TabAlignment.start`.
This change also removes the `52.0` pixel offset.
The following code snippets show how to use `TabBar.tabAlignment` to align the tabs
to the start of the scrollable `TabBar`:
```

**Review by atsansone — CHANGES_REQUESTED** [review]
@TahaTesser : A few changes for length and semantic breaks. Thanks!

**Comment by TahaTesser** [comment]
@atsansone 
Appreciate the review, made the requested changes

File: src/release/breaking-changes/tab-alignment.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+* `TabAlignment.center`
+
+The fixed `TabBar` supports the following alignments:
+
+* `TabAlignment.fill`
+* `TabAlignment.center`
+
+When you set `ThemeData.useMaterial3` to `true`,
+a scrollable `TabBar` aligns tabs as `TabAlignment.startOffset` by default.
+To change this alignment, set the `TabBar.tabAlignment` property.
```
Comment (Piinks) [inline src/release/breaking-changes/tab-alignment.md:33]: You can also use the theme, correct?

Comment (TahaTesser) [reply src/release/breaking-changes/tab-alignment.md:33]: That's right.

**Review by atsansone — APPROVED** [review]
@TahaTesser : I'll approve this as is. Keep in mind for next time that we hard wrap lines at 80 characters. Thanks!

**Pull request merged**